packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L1ETHGateway.sol

```diff
@@ -174,4 +174,21 @@ contract OVM_L1ETHGateway is iOVM_L1ETHGateway, OVM_CrossDomainEnabled, Lib_Addr
         (bool success, ) = _to.call{value: _value}(new bytes(0));
         require(success, 'TransferHelper::safeTransferETH: ETH transfer failed');
     }
+
+    /*****************************
+     * Temporary - Migrating ETH *
+     *****************************/
+
+    /**
+     * @dev migrates entire ETH balance to another gateway
+     * @param _to Gateway Proxy address to migrate ETH to
+     */
+    function migrateEth(address payable _to) external {
+        address owner = Lib_AddressManager(libAddressManager).owner();
+        require(msg.sender == owner, "Only the owner can migrate ETH");
+        uint256 balance = address(this).balance;
+        OVM_L1ETHGateway(_to).donateETH{value:balance}();
+    }
+
+    function donateETH() external payable {}
```

Comment:
What is this used for?

Reply:
for the next Gateway to be able to receive the ETH - used this instead of the default `receive` to make it clear that this is *not* a valid way to deposit ETH into L2

Reply:
So, it's a placeholder for future functionality? 
Maybe it should revert for now?
Could you please add a docstring to it?

Reply:
`donateEth` is just a payable function that is used here: `OVM_L1ETHGateway(_to).donateETH{value:balance}();`
to receive the migrated ETH.